installer/default_platform.conf

```diff
@@ -469,9 +539,34 @@ EOF
         fi
     fi
 
+# Make a first grub config file that located in default debian path:/boot/efi/EFI/debian/
+# this first grub.cfg will call the complete grub.cfg created below with sonic configuration
+tmp_config=$(mktemp)
+cat <<EOF > $tmp_config
+search --no-floppy --label --set=root $demo_volume_label
+set prefix=(\$root)'/grub'
+configfile \$prefix/grub.cfg
+EOF
+
+# Copy first grub.cfg as entrypoint to default debian path where grubx64.efi expected it.
+mkdir -p /boot/efi/EFI/debian/
+echo "cp $tmp_config /boot/efi/EFI/debian/grub.cfg"
+cp $tmp_config /boot/efi/EFI/debian/grub.cfg
+
 # Add extra linux command line
 echo "EXTRA_CMDLINE_LINUX=$extra_cmdline_linux"
 GRUB_CMDLINE_LINUX="$GRUB_CMDLINE_LINUX $extra_cmdline_linux"
+GRUB_CFG_LINUX_CMD=""
+GRUB_CFG_INITRD_CMD=""
+if [ "$firmware" = "uefi" ] ; then
```

Comment:
what about the bios is uefi, but do not support secure boot?

Reply:
Good comment, 
first of all, it will work, `linuxefi` command it's a command similar than `linux` to load the kernel from grub.
The different it's that `linuxefi` is supported by UEFI system (when secure boot enabled or disabled).
The `linux` command is not supporting some new feature that exist in UEFI like Secure Boot.
So, in general it should be better to always use this command in UEFI systems, and this also give the possibility to move from secure boot enable and disable with flexibility.